src/tests/IntegrationTests/Events/RepositoryManagerTests.cs

```diff
@@ -504,7 +505,64 @@ await RepositoryManager.SwitchBranch("branch2")
             repositoryManagerListener.DidNotReceive().OnLocksUpdated(Args.EnumerableGitLock);
         }
 
-        private void WaitForNotBusy(RepositoryManagerEvents managerEvents, int seconds = 1)
+		[Test]
+		public async Task ShouldCheckoutFiles()
+		{
+			await Initialize( TestRepoMasterCleanSynchronized );
+
+			var repositoryManagerListener = Substitute.For<IRepositoryManagerListener>();
+			repositoryManagerListener.AttachListener( RepositoryManager, repositoryManagerEvents );
+
+			var expected = new GitStatus
+			{
+				Behind = 1,
+				LocalBranch = "master",
+				RemoteBranch = "origin/master",
+				Entries =
+					new List<GitStatusEntry> {
+						new GitStatusEntry("foobar.txt", TestRepoMasterCleanSynchronized.Combine("foobar.txt"),
+							"foobar.txt", GitFileStatus.Untracked)
+					}
+			};
+
+			var result = new GitStatus();
+			Environment.Repository.OnStatusUpdated += status => { result = status; };
+
+			var foobarTxt = TestRepoMasterCleanSynchronized.Combine( "foobar.txt" );
+			foobarTxt.WriteAllText( "foobar" );
+
+			await TaskManager.Wait();
+			RepositoryManager.WaitForEvents();
+			WaitForNotBusy( repositoryManagerEvents, 1 );
+
+			repositoryManagerListener.Received().OnStatusUpdate( Args.GitStatus );
+			repositoryManagerListener.DidNotReceive().OnActiveBranchChanged( Arg.Any<ConfigBranch?>() );
+			repositoryManagerListener.DidNotReceive().OnActiveRemoteChanged( Arg.Any<ConfigRemote?>() );
+			repositoryManagerListener.DidNotReceive().OnLocalBranchListChanged();
+			repositoryManagerListener.DidNotReceive().OnRemoteBranchListChanged();
+			repositoryManagerListener.Received().OnIsBusyChanged( Args.Bool );
+			repositoryManagerListener.DidNotReceive().OnLocksUpdated( Args.EnumerableGitLock );
```

Comment:
Here we are showing that our `RepositoryManager` has fired a `StatusUpdate` event and a `BusyChanged` event, both at least once.

So here what you should do is check that the git status is what you expect.
You are doing it further below on line 562.

Add the following code here...

```
result.AssertEqual(expected);

repositoryManagerListener.ClearReceivedCalls();
repositoryManagerEvents.Reset();
```

That way we can check the git status is what you expected.
And we can reset the `repositoryManagerListener` for further testing.

After you fix this, the test starts failing and we can work on figuring that out next.

Reply:
What should the `GitFileStatus` of `foobar.txt` be after checkout?
It is currently set to `GitFileStatus.Untracked`, should it be `GitFileStatus.None` instead?

Reply:
I was going to talk about that... but I didn't want to overload you with information...
After you make and push that change I mentioned.
The test will fail and I will continue lecturing then.. :)